healthcheck.tf

```
@@ -0,0 +1,58 @@
+locals {
+  healthchecks = module.this.enabled && var.healthchecks != null ? {
+    for healthcheck in flatten(var.healthchecks) :
+    format("%s-%s-%s",
+      lookup(healthcheck, "name", null) == null ? module.this.id : healthcheck.name,
+      healthcheck.type,
+      healthcheck.address
+    ) => healthcheck
+  } : {}
+}
+
+resource "time_sleep" "wait_for_records_creation" {
+  count = module.this.enabled ? 1 : 0
+
+  create_duration = "60s"
+
+  depends_on = [
+    cloudflare_record.default
+  ]
+}
+
+resource "cloudflare_healthcheck" "default" {
+  for_each = local.healthchecks
+
+  zone_id                      = local.zone_id
+  name                         = lookup(each.value, "name", null) == null ? each.key : each.value.name
```

This confused me at first, as to why you didn't just use ` lookup(each.value, "name", each.key)` 
```suggestion
  zone_id = local.zone_id
  # Cover both the case where the map is missing the key and where they key is present but the value is null
  name                         = lookup(each.value, "name", null) == null ? each.key : each.value.name
```